apps/api/v2/test/oauth.e2e-spec.ts

```diff
@@ -1,81 +1,312 @@
 import { AppModule } from "@/app.module";
+import { HttpExceptionFilter } from "@/filters/http-exception.filter";
+import { PrismaExceptionFilter } from "@/filters/prisma-exception.filter";
 import { AuthModule } from "@/modules/auth/auth.module";
 import { NextAuthStrategy } from "@/modules/auth/strategy";
+import { CreateOAuthClientInput } from "@/modules/oauth/input/create-oauth-client";
+import { UpdateOAuthClientInput } from "@/modules/oauth/input/update-oauth-client";
 import { OAuthClientModule } from "@/modules/oauth/oauth-client.module";
+import { PrismaModule } from "@/modules/prisma/prisma.module";
 import { UserModule } from "@/modules/repositories/user/user-repository.module";
 import { INestApplication } from "@nestjs/common";
+import { NestExpressApplication } from "@nestjs/platform-express";
 import { Test } from "@nestjs/testing";
+import { Membership, PlatformOAuthClient, Team, User } from "@prisma/client";
 import * as request from "supertest";
 
+import { SUCCESS_STATUS } from "@calcom/platform-constants";
+import { ApiSuccessResponse } from "@calcom/platform-types";
+
+import { bootstrap } from "../src/app";
+import { MembershipFixtures } from "./fixtures/membership.fixtures";
+import { TeamFixtures } from "./fixtures/team.fixtures";
+import { UsersFixtures } from "./fixtures/users.fixtures";
 import { NextAuthMockStrategy } from "./mocks/next-auth-mock.strategy";
 import { withNextAuth } from "./utils/withNextAuth";
 
-describe("oAuth: User is not authenticated", () => {
-  let app: INestApplication;
+describe("OAuth Client Endpoints", () => {
+  describe("User Not Authenticated", () => {
+    let appWithoutAuth: INestApplication;
 
-  beforeAll(async () => {
-    const moduleRef = await Test.createTestingModule({
-      imports: [AppModule, OAuthClientModule],
-    }).compile();
+    beforeAll(async () => {
+      const moduleRef = await Test.createTestingModule({
+        providers: [PrismaExceptionFilter, HttpExceptionFilter],
+        imports: [AppModule, OAuthClientModule, UserModule, AuthModule, PrismaModule],
+      }).compile();
+      appWithoutAuth = moduleRef.createNestApplication();
+      bootstrap(appWithoutAuth as NestExpressApplication);
+      await appWithoutAuth.init();
+    });
 
-    app = moduleRef.createNestApplication();
-    await app.init();
-  });
+    it(`/GET`, () => {
+      return request(appWithoutAuth.getHttpServer()).get("/api/v2/oauth-clients").expect(401);
+    });
+    it(`/GET/:id`, () => {
+      return request(appWithoutAuth.getHttpServer()).get("/api/v2/oauth-clients/1234").expect(401);
+    });
+    it(`/POST`, () => {
+      return request(appWithoutAuth.getHttpServer()).post("/api/v2/oauth-clients").expect(401);
+    });
+    it(`/PUT/:id`, () => {
+      return request(appWithoutAuth.getHttpServer()).put("/api/v2/oauth-clients/1234").expect(401);
+    });
+    it(`/delete/:id`, () => {
```

Comment:
Nit pick - in all tests we have `/delete` in lower case instead of upper case like `/PUT` so would be good to have `/DELETE`